pywatts/modules/postprocessing/ensemble.py

```diff
@@ -166,6 +129,45 @@ def transform(self, **kwargs) -> xr.DataArray:
         if not all(all(index) == all(list_of_indexes[0]) for index in list_of_indexes):
             raise ValueError("The indexes of the given time series for averaging do not match")
 
-        result = np.average(list_of_series, axis=0, weights=self.weights)
+        result = np.average(list_of_series, axis=0, weights=self._weights)
 
         return numpy_to_xarray(result, series, self.name)
+
+    def _calculate_loss(self, ps, ts):
+
+        t_ = np.array([t.values for t in ts.values()])
+        loss_values = []
+        for p in ps.values():
+            p_ = p.values
+            if self.loss_metric == "rmse":
+                loss_values.append(np.sqrt(np.mean((p_ - t_) ** 2)))
+            elif self.loss_metric == "mae":
+                loss_values.append(np.mean(np.abs((p_ - t_))))
+            else:
+                WrongParameterException(
+                    "The specified loss metric is not implemented.",
+                    "Make sure to pass the loss metric 'rmse' or 'mae'.",
+                    self.name
+                )
+
+        return loss_values
+
+    def _drop_forecasts(self, loss: list):
+        loss_values_dropped = []
+        if self.k_best is not None:
+            # Do not sort the loss_values! Otherwise, the weights do not match the given forecasts.
+            if self.k_best == "auto":
+                q75, q25 = np.percentile(loss, [75, 25])
+                iqr = q75 - q25
+                upper_bound = q75 + 1.5 * iqr  # only check for outliers with high loss
+                loss_values_dropped = [value for value in loss if not (value <= upper_bound)]
+            elif self.k_best > len(loss):
+                raise WrongParameterException(
+                    "The given k is greater than the number of the given loss values.",
+                    f"Make sure to define k <= {len(loss)}.",
+                    self.name
+                )
+            else:
+                loss_values_dropped = sorted(loss)[self.k_best:]
```

Comment:
In line 158 you write that it is not allowed to sort the loss. So why it is allowed here?